src/middlewared/middlewared/plugins/crypto_/__init__.py

```diff
@@ -15,7 +16,7 @@ async def setup(middleware):
         # create a self signed cert if it doesn't exist and set ui_certificate to it's value
         try:
             if not any('freenas_default' == c['name'] for c in certs):
-                cert, key = await middleware.call('cryptokey.generate_self_signed_certificate')
+                cert, key = generate_self_signed_certificate()
```

Comment:
async method calling a sync method that calls other sync methods.

Reply:
In 99.99% of cases it would only be called once during TrueNAS setup ( not middleware lifecycle ) and take less then a second to complete. Is it problematic ?

Reply:
Formally speaking, I think it performs I/O (consuming random seed) so it might block the loop? What's the need to move this outside plugins anyway?